kakao/src/main/kotlin/com/agoda/kakao/delegate/DataInteractionDelegate.kt

```diff
@@ -0,0 +1,71 @@
+package com.agoda.kakao.delegate
+
+import android.view.View
+import androidx.test.espresso.*
+import androidx.test.espresso.action.AdapterViewProtocol
+import androidx.test.espresso.assertion.ViewAssertions
+import com.agoda.kakao.Kakao
+import com.agoda.kakao.intercept.Interceptor
+import com.agoda.kakao.screen.Screen
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers
+import org.jetbrains.annotations.Contract
+
+/**
+ * Delegation class for [DataInteraction].
+ * Wraps all available public calls and intercepts [check] and [perform].
+ *
+ * @see Delegate
+ * @see Interceptor
+ */
+class DataInteractionDelegate(override var interaction: DataInteraction) : Delegate<DataInteraction, ViewAssertion, ViewAction> {
+    override var interceptor: Interceptor<DataInteraction, ViewAssertion, ViewAction>? = null
+
+    @Contract("_->this")
+    fun atPosition(atPosition: Int) = this.also {
+        interaction.atPosition(atPosition)
+    }
+
+    fun check(viewAssert: ViewAssertion): ViewInteractionDelegate {
+        return ViewInteractionDelegate(
+                if (!interceptCheck(viewAssert)) {
+                    interaction.check(viewAssert)
+                } else {
+                    interaction.check(ViewAssertions.matches(Matchers.anything()))
```

Comment:
We still need to return `ViewInteraction` transformed from `DataInteraction`, but since the real call has been overridden, we just silently do an always true check to do the transformation. Or else I'll have to use heavy reflection here, which is not desirable.